tests/combine_span_test.cpp

```diff
@@ -0,0 +1,280 @@
+  // cases where the two update rules differ
+  constexpr auto direction = TypeParam::value;
+
+  span_test::mock_rng rng{0.5};
+
+  nuts::SpanW<double> span_from = span_test::dummy_span(1.8);
+  nuts::SpanW<double> span_to = span_test::dummy_span(0.3);
+  double new_logp = nuts::log_sum_exp(span_from.logp_, span_to.logp_);
+
+  // metropolis: 0.3 - 1.8 = -1.5 < -0.3, rejects
+  {
+    nuts::SpanW<double> span_old = span_from;
+    nuts::SpanW<double> span_new = span_to;
+    nuts::SpanW<double> combined =
+        nuts::combine<nuts::Update::Metropolis, direction>(
+            rng, std::move(span_old), std::move(span_new));
+
+    EXPECT_SELECTED_POINT_EQUAL(combined, span_from, new_logp);
+  }
+
+  // barker: 0.3 - lsexp(0.3, 1.8) = 0.3 - 2.00 = -2.00 < -0.3, rejects
+  {
+    nuts::SpanW<double> span_old = span_from;
+    nuts::SpanW<double> span_new = span_to;
+    nuts::SpanW<double> combined =
+        nuts::combine<nuts::Update::Barker, direction>(rng, std::move(span_old),
+                                                       std::move(span_new));
+
+    EXPECT_SELECTED_POINT_EQUAL(combined, span_from, new_logp);
+  }
+}
+
+TYPED_TEST(CombineSpansSymmetric, BarkerAndMetropolisBothAccept) {
+  // cases where the two update rules differ
+  constexpr auto direction = TypeParam::value;
+
+  span_test::mock_rng rng{0.5};
+  nuts::SpanW<double> span_from = span_test::dummy_span(0.3);
+  nuts::SpanW<double> span_to = span_test::dummy_span(1.8);
+  double new_logp = nuts::log_sum_exp(span_from.logp_, span_to.logp_);
+
+  // metropolis: 1.8 - 0.3 = 1.5 > -0.3, accepts
+  {
+    nuts::SpanW<double> span_old = span_from;
+    nuts::SpanW<double> span_new = span_to;
+    nuts::SpanW<double> combined =
+        nuts::combine<nuts::Update::Metropolis, direction>(
+            rng, std::move(span_old), std::move(span_new));
+
+    EXPECT_SELECTED_POINT_EQUAL(combined, span_to, new_logp);
+  }
+
+  // barker: 1.8 - lsexp(0.3, 1.8) = 1.8 - 2.0 = -.2 > -0.3, accepts
+  {
+    nuts::SpanW<double> span_old = span_from;
+    nuts::SpanW<double> span_new = span_to;
+    nuts::SpanW<double> combined =
+        nuts::combine<nuts::Update::Barker, direction>(rng, std::move(span_old),
+                                                       std::move(span_new));
+
+    EXPECT_SELECTED_POINT_EQUAL(combined, span_to, new_logp);
+  }
+}
+
+// The following test uses the same logic for all combinations of
+// direction and update type
+template <typename T>
+class CombineSpansUniversal : public ::testing::Test {};
+
+using AllCombinations =
+    ::testing::Types<span_test::ForwardsMetropolis,
+                     span_test::BackwardsMetropolis, span_test::ForwardsBarker,
+                     span_test::BackwardsBarker>;
+TYPED_TEST_SUITE(CombineSpansUniversal, AllCombinations);
+
+TYPED_TEST(CombineSpansUniversal, SelectedPositionAccepted) {
+  // test that the 'selected' position and gradient are as expected
```

Comment:
Why is "selected" quoted here?  

Can we just name the tests something more meaningful and drop the doc?